### DISCUSSION TITLE
Yarp adds a trailing slash to destination

### DISCUSSION THREAD

**Discussion by czobrisky** [dotnet/yarp#1536] [Q&A] [answered]
We've configured destinations and yarp is adding a trailing slash to it. Why is it making this assumption and can we stop it from happening? 
Eg.

`http://localhost/path` is converted to `http://localhost/path/`.

**Comment by Tratcher** [comment, marked as answer]
The proxied request path is the combination of the destination address and the original request path. In your example the request path is "/", and the destination address is "http://localhost/path", so you get "http://localhost/path/".

You can adjust this with a request transform:
https://microsoft.github.io/reverse-proxy/articles/transforms.html#addrequesttransform

**Reply by czobrisky** [reply]
Even if my request is sent to "https://localhost:8080", it seems somewhere a "/" is being added. Is that in Kestrel or YARP or is Invoke-WebRequest adding a "/"?

**Reply by Tratcher** [reply]
In HTTP the Path is never empty. If you look at it on the wire, https://localhost:8080 is sent as:
```
GET / HTTP/1.1
Host: localhost:8080
```

**Reply by czobrisky** [reply]
Thank you for clarifying.
